CMakeLists.txt

```
@@ -113,14 +114,34 @@ ENDIF ()
 
 ## Flags
 
-### Warnings
+### C++ 17/20 support
 
 IF (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
 
-    IF (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 13.0)
+    IF (BUILD_WITH_CXX_17)
 
-        MESSAGE (FATAL_ERROR "GCC version must be at least 13.0")
+        SET (CMAKE_CXX_STANDARD 17)
 
+        MESSAGE (STATUS "C++17 support enabled")
+
+        IF (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.0)
+
+            MESSAGE (FATAL_ERROR "GCC version must be at least 7.0")
+
+        ENDIF ()
+    
+    ELSE ()
+
+        MESSAGE (STATUS "C++20 support enabled")
+
+        SET (CMAKE_CXX_STANDARD 20)
+
+        IF (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 13.0)
+
+            MESSAGE (FATAL_ERROR "GCC version must be at least 13.0")
```

```suggestion
            MESSAGE (FATAL_ERROR "GCC version must be at least 13.0 to build with C++20")
```

nit: mostly to give a hint that there are options besides C++20 for older GCC versions